src/components/application_manager/rpc_plugins/rc_rpc_plugin/src/rc_pending_resumption_handler.cc

```diff
@@ -0,0 +1,292 @@
+    queue_freezed.pop();
+    if (queue_freezed.empty()) {
+      freezed_resumptions_.erase(it_queue_freezed);
+    }
+    return freezed_resumption;
+  };
+
+  auto freezed_resumption = pop_front_freezed_resumptions(module_uid);
+  if (!freezed_resumption) {
+    LOG4CXX_DEBUG(logger_, "Not freezed resumptions found");
+    return;
+  }
+
+  auto& resumption_request = *freezed_resumption;
+  auto subscription_request =
+      std::make_shared<smart_objects::SmartObject>(resumption_request.message);
+  const auto fid = GetFunctionId(subscription_request);
+  const auto cid =
+      resumption_request
+          .message[app_mngr::strings::params][app_mngr::strings::correlation_id]
+          .asInt();
+  LOG4CXX_DEBUG(logger_,
+                "Subscribing for event with function id: "
+                    << fid << " correlation id: " << cid);
+  subscribe_on_event(fid, cid);
+  pending_requests_.emplace(cid, *subscription_request);
+  LOG4CXX_DEBUG(logger_,
+                "Sending request with fid: " << fid << " and cid: " << cid);
+  application_manager_.GetRPCService().ManageHMICommand(subscription_request);
+}
+
+smart_objects::SmartObjectSPtr
+RCPendingResumptionHandler::CreateSubscriptionRequest(
+    const ModuleUid& module, const uint32_t correlation_id) const {
+  auto request =
+      RCHelpers::CreateUnsubscribeRequestToHMI(module, correlation_id);
+  auto& msg_params = (*request)[application_manager::strings::msg_params];
+  msg_params[message_params::kSubscribe] = true;
+  return request;
+}
+
+void RCPendingResumptionHandler::RaiseEvent4Response(
+    const smart_objects::SmartObject& subscription_response,
+    const uint32_t correlation_id) const {
+  smart_objects::SmartObject event_message = subscription_response;
+  event_message[app_mngr::strings::params][app_mngr::strings::correlation_id] =
+      correlation_id;
+
+  smart_objects::SmartObject& module_data =
+      event_message[app_mngr::strings::msg_params][message_params::kModuleData];
+  if (module_data.keyExists(message_params::kAudioControlData)) {
+    smart_objects::SmartObject& audio_control_data =
+        module_data[message_params::kAudioControlData];
+    if (audio_control_data.keyExists(message_params::kKeepContext)) {
+      audio_control_data.erase(message_params::kKeepContext);
+    }
+  }
+
+  app_mngr::event_engine::Event event(
+      hmi_apis::FunctionID::RC_GetInteriorVehicleData);
+  event.set_smart_object(event_message);
+  event.raise(application_manager_.event_dispatcher());
+}
+
+const hmi_apis::FunctionID::eType RCPendingResumptionHandler::GetFunctionId(
+    smart_objects::SmartObjectSPtr subscription_request) const {
+  auto& request_ref = *subscription_request;
+  const auto function_id = static_cast<hmi_apis::FunctionID::eType>(
+      request_ref[app_mngr::strings::params][app_mngr::strings::function_id]
+          .asInt());
+
+  return function_id;
+}
+
+ModuleUid RCPendingResumptionHandler::GetModuleUid(
```

Comment:
Dont we have such in RCHelpers?

Reply:
@LuxoftAKutsan 
Not present in RCHelpers